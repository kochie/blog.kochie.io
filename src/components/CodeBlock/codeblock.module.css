.highlight-code-line-dark {
  background-color: #022a4b;
  display: block;
  /* width: 100%; */
  padding-right: 1.3125rem;
  padding-left: 1.0625rem;
  border-left: 0.25rem solid #ffa7c4;
  margin-left: -1.3125rem;
  margin-right: -1.3125rem;
}

.highlight-code-line-light {
  background-color: lightgrey;
  display: block;
  padding-right: 1.3125rem;
  padding-left: 1.0625rem;
  border-left: 0.25rem solid #ffa7c4;
  margin-left: -1.3125rem;
  margin-right: -1.3125rem;
}

@font-face {
  font-family: 'Cascadia Code PL';
  src: url(../../assets/fonts/CascadiaCodePL.ttf);
  font-display: swap;
}

.code {
  padding: 1.3125rem;
  font-family: 'Cascadia Code PL', monospace;
  tab-size: 2;
  overflow: auto;
  font-size: small;

  @media (--medium-viewport) and (orientation: landscape) {
    font-size: smaller;
  }

  @media (--large-viewport) and (orientation: landscape) {
    font-size: unset;
  }
}

.wrap {
  white-space: pre-wrap;
  word-break: break-word;
}
