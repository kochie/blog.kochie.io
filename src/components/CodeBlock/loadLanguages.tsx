import { Prism } from 'prism-react-renderer'
import CodeBlock, { CodeBlockProps } from './codeblock'
import { PropsWithChildren, Suspense } from 'react'
;(typeof global !== 'undefined' ? global : window).Prism = Prism

export default async function LoadWrapper(
  props: PropsWithChildren<CodeBlockProps>
) {
  return (
    <Suspense fallback={<div>Loading languages</div>}>
      <CodeBlock {...props} />
    </Suspense>
  )
}
