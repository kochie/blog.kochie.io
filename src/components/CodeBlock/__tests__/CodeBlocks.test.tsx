import React, { Suspense } from 'react'
import { render } from '@testing-library/react'
import { describe, test, expect } from 'vitest'
import CodeBlock from '@/components/CodeBlock/codeblock'

describe('CODEBLOCKS COMPONENT', () => {
  test('renders correctly', async () => {
    const { asFragment } = render(
      <Suspense>
        <CodeBlock className={'language-typescript{10}'} />
      </Suspense>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('renders highlight', async () => {
    const { asFragment } = render(
      <Suspense>
        <CodeBlock className={'language-typescript{10}'} />
      </Suspense>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('renders highlight range', async () => {
    const { asFragment } = render(
      <Suspense>
        <CodeBlock className={'language-typescript{10-15}'} />
      </Suspense>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('renders line numbers', async () => {
    const { asFragment } = render(
      <Suspense>
        <CodeBlock className={'language-typescript{}[LineNumbers]'} />
      </Suspense>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
