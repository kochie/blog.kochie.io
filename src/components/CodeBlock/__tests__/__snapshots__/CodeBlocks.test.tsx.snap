// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CODEBLOCKS COMPONENT > renders correctly 1`] = `<DocumentFragment />`;

exports[`CODEBLOCKS COMPONENT > renders highlight 1`] = `<DocumentFragment />`;

exports[`CODEBLOCKS COMPONENT > renders highlight range 1`] = `<DocumentFragment />`;

exports[`CODEBLOCKS COMPONENT > renders line numbers 1`] = `<DocumentFragment />`;
